maas/resource_maas_logical_volume.go

```diff
@@ -0,0 +1,219 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASLogicalVolume() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMAASLogicalVolumeCreate,
+		ReadContext:   resourceMAASLogicalVolumeRead,
+		UpdateContext: resourceMAASLogicalVolumeUpdate,
+		DeleteContext: resourceMAASLogicalVolumeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"fs_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The file system type (e.g. `ext4`). If this is not set, the volume is unformatted.",
+			},
+			"machine": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The machine identifier (system ID, hostname, or FQDN) that owns the volume group.",
+			},
+			"mount_options": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The options used for the logical volume mount.",
+			},
+			"mount_point": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The mount point used. If this is not set, the logical volume is not mounted.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name for this logical volume",
+			},
+			"size_gigabytes": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
```

Comment:
This `ForceNew` should be removed. Most probably it is a remnant from previous commits when the gomaasclient in use could not properly set the size